### PR TITLE
fix(aws-apigateway): add optional property to spec-restapi to be able to enable compression

### PR DESCRIPTION
**Pull request by virajmavani**
This PR addresses the feature request: https://github.com/aws/aws-cdk/issues/22926

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts [hunk 1]
```diff
@@ -261,6 +261,18 @@ export interface SpecRestApiProps extends RestApiBaseProps {
...
+   * A nullable integer that is used to enable compression (with non-negative
+   * between 0 and 10485760 (10M) bytes, inclusive) or disable compression
+   * (when undefined) on an API. When compression is enabled, compression or
+   * decompression is not applied on the payload if the payload size is
+   * smaller than this value. Setting it to zero allows compression for any
+   * payload size.
+   *
+   * @default - Compression is disabled.
+   */
+  readonly minimumCompressionSize?: number;
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-apigateway/lib/restapi.ts:275]: Given the limitations, could you please add validation that the number provided must be in this range? Please also add unit tests (both in range and out of range) to validate this.

Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-apigateway/lib/restapi.ts:275]: I'm also thinking that this should be moved into the base props instead of adding it individually to both `RestApiProps` and `SpecRestApiProps`.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
